api/license.go

```diff
@@ -45,11 +45,11 @@ func addLicense(c *Context, w http.ResponseWriter, r *http.Request) {
 	fileData := fileArray[0]
 
 	file, err := fileData.Open()
-	defer file.Close()
```

Comment:
Panics due to a nil deference if `fileData.Open()` returned an error instead of a file. Fixing several instances of this mistake here.